### PR TITLE
Refl-cpp recipe

### PR DESCRIPTION
**Pull request by rilerez**
Specify library name and version:  *refl-cpp/0.12.1**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): It is a library someone I know wanted to use from a Conan project

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by rilerez** [comment]
This library requires at least C++17. It does not currently have a CMakeLists.txt specifying this. Should I add this information to conan somehow? Should I just make a CMakeLists? if so, is it better to put that here or in the original repository?

**Comment by prince-chrismc** [comment]
You are adding two recipes in this PR... Can you please remove the _old_ one?


> This library requires at least C++17. It does not currently have a CMakeLists.txt specifying this. Should I add this information to conan somehow? Should I just make a CMakeLists? if so, is it better to put that here or in the original repository?

The easiest way to do this is with copy and paste from conan-io/conan#8002 which is our usually was of doing it... there's lots of C++17 examples in CCI if you need to compiler versions

**Comment by rilerez** [comment]
Thanks! Is the easiest way to remove the old one to close this and try again?

**Comment by prince-chrismc** [comment]
> Thanks! Is the easiest way to remove the old one to close this and try again?

If you are not familiar with git, I'd suggest using the file explorer to remove the folder and run `git commit -a -m "remove old"` and a usually `git push` to fix it.

Hopefully that helps

File: recipes/refl-cpp/all/test_package/example.cpp [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#include "refl.hpp"
+
+int main() {
+    return 0;
```
Comment (prince-chrismc) [inline recipes/refl-cpp/all/test_package/example.cpp:4]: Can you please add a line to all a method or ctor to test the linkage (or make sure theres no missing deps)

File: recipes/refl-cpp/all/conanfile.py [hunk 2]
```diff
@@ -11,6 +11,11 @@ class ReflCppConan(ConanFile):
     description = "A modern compile-time reflection library for C++ with support for overloads, templates, attributes and proxies "
     topics = ("header", "header-only", "reflection", "modern", "metaprogramming")
     no_copy_source = True
+    settings = "os", "compiler", "build_type", "arch"
```
Comment (prince-chrismc) [inline recipes/refl-cpp/all/conanfile.py:14]: Seems to be the only one used

```suggestion
    settings =  "compiler"
```

**Review by prince-chrismc — COMMENTED** [review]
Very close

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — COMMENTED** [review]
Thank you for your contribution. Please, take a look on my review.

File: recipes/refl-cpp/all/test_package/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import os
+from conans import ConanFile, CMake, tools
+
+
+class TestPackageConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
```
[the remark this answers is not in the document]
Comment (rilerez) [reply recipes/refl-cpp/all/test_package/conanfile.py:7]: What is cmake_find_package_multi for?

[the remark this answers is not in the document]
Comment (prince-chrismc) [reply recipes/refl-cpp/all/test_package/conanfile.py]: To generate the files need for using `find_package` (to test the official upstream way)

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

**Review by czoido — APPROVED** [review]

**Pull request merged**